src/server.ts

```diff
@@ -72,57 +89,77 @@ app.post("/movies", async (request: Request, response: Response) => {
 });
 
 app.patch("/movies/:id", async (request: Request, response: Response) => {
+  const client = await MongoClient.connect(DB_URL);
+  const db = client.db("moviesDB");
+  const collection = db.collection("movies");
+
   let id: string = request.params.id;
   const { comment, personalScore } = request.body;
 
-  const movie = movieStore.findById(id);
+  const movie = await collection.findOne({ imdbID: id });
 
   if (!movie) {
     return response.status(404).json({
       message: "Movie not found",
     });
   }
 
+  const updateFields: { [k: string]: any } = {};
   if (comment) {
-    movie["comment"] = comment;
+    updateFields["comment"] = comment;
   }
   if (personalScore) {
-    movie["personalScore"] = personalScore;
+    updateFields["personalScore"] = personalScore;
   }
 
+  await collection.updateOne({ imdbID: id }, { $set: updateFields });
+  console.log("MOVIE", movie);
   return response.status(200).json({
     message: `Movie updated: ${movie.Title}`,
     data: movie,
   });
 });
 
-app.delete("/movies/:id", (request: Request, response: Response) => {
+app.delete("/movies/:id", async (request: Request, response: Response) => {
+  const client = await MongoClient.connect(DB_URL);
```

Comment:
You create new mongodb client for every single endpoint and for every request, it's not good.
 ```
  const client = await MongoClient.connect(DB_URL);
  const db = client.db("moviesDB");
  const collection = db.collection("movies");
  ```
Move this code to line 16, and then just call `collection` when you need it.

Reply:
@rfist, won't work because of 
```
const client = await MongoClient.connect(DB_URL);
```